ui/js/controllers/perf/alerts.js

```diff
@@ -12,10 +12,35 @@ perf.factory('PhBugs', [
                         return alert.status !== phAlertStatusMap.INVALID;
                     });
                     var template = response.data[0];
+                    function getAlertSummary(alertSummary) {
+                        var improved = "";
+                        var regressed = "";
+                        for (var i = 0; i < alertSummary.alerts.length; i++) {
+                            if (alertSummary.alerts[i].amount_abs < 0 && alertSummary.alerts[i].visible) {
```

Comment:
You can't say whether something is better or worse based on amount_abs. You should use the is_regression property.
Also, you could probably express this more concisely by aggregating the alerts into improvement / regression categories, then doing a map/join on the groups.

E.g. (omitting some details):
```js
var resultStr = "";
var improved = _.filter(alertSummary.alerts, function(alert) { return !alert.is_regression && alert.visible; });
var regressed = _.filter(alertSummary.alerts, function(alert) { return alert.is_regression && alert.visible; });
if (improved.length > 0) {
   resultStr += "Summary of tests that improved:\n" + _.map(improved, function(alert) { return alert.title + '-' + ... }).join('\n') + "\n";
}
...
return resultStr;
```